library/src/main/java/com/owncloud/android/lib/resources/files/SearchRemoteOperation.java

```diff
@@ -169,5 +175,82 @@ protected RemoteOperationResult<List<RemoteFile>> run(OwnCloudClient client) {
         }
         return result;
     }
+
+    @Override
+    public RemoteOperationResult run(NextcloudClient client) {
+        RemoteOperationResult<List<RemoteFile>> result;
+        NcSearchMethod searchMethod = null;
+
+        String webDavUrl = client.getDavUri().toString();
+
+        SearchInfo searchInfo = new SearchInfo("NC",
+                Namespace.XMLNS_NAMESPACE,
+                searchQuery);
+
+        try {
+            searchMethod = new NcSearchMethod(webDavUrl,
+                    searchInfo,
+                    searchType,
+                    getClientNew().getUserIdPlain(),
+                    timestamp,
+                    limit,
+                    filterOutFiles,
+                    capability,
+                    startDate,
+                    endDate);
+
+            // disable redirect
+            OkHttpClient disabledRedirectClient = client.getClient()
+                    .newBuilder()
+                    .followRedirects(false)
+                    .authenticator(new NextcloudAuthenticator(client.getCredentials(), "Authorization"))
+                    .build();
+
+            Document searchDocument = searchMethod.getDocumentQuery(searchInfo);
+            String searchString = transformDocumentToString(searchDocument);
+
+            ArrayList<Response> responses = new ArrayList<>();
+
+            new DavResource(
+                    disabledRedirectClient,
+                    HttpUrl.get(client.getDavUri().toString()))
+                    .search(searchString, (response, hrefRelation) -> {
+                        responses.add(response);
+                    });
+
+            // get data from remote folder
+            WebDavFileUtils webDavFileUtils = new WebDavFileUtils();
+
+            ArrayList<RemoteFile> list = new ArrayList<>();
+            for (Response response : responses) {
+                list.add(webDavFileUtils.parseResponse(response, client.getFilesDavUri()));
+            }
+
+            // Result of the operation
+            result = new RemoteOperationResult<>(
+                    true,
+                    HttpStatus.SC_OK,
+                    searchMethod.getResponseHeaders());
+            // Add data to the result
+            if (result.isSuccess()) {
+                result.setResultData(list);
+            }
+        } catch (Exception e) {
+            result = new RemoteOperationResult<>(e);
+        } finally {
+            if (searchMethod != null) {
+                searchMethod.releaseConnection();  // let the connection available for other methods
+            }
```

Comment:
This does not seem necessary, as the search method is never called (only used to get the query string)